File: Quiz-App-StarterKit/Source code v9.2 Turn based/Classes/Utilities/UILabel+VerticalAlign.m
//
//  UILabel+Alignment.m
//  QUIZ_APP
//
//  Created by Satish Nerlekar on 13/08/12.
//  Copyright (c) 2012 Heaven Apps. All rights reserved.
//

#import "UILabel+VerticalAlign.h"

@implementation UILabel (VerticalAlign)
- (void)alignTop {
    
    NSDictionary *attributes = @{ NSFontAttributeName: self.font};
    CGSize fontSize = [self.text sizeWithAttributes:attributes];
    double finalHeight = fontSize.height * self.numberOfLines;
    double finalWidth = self.frame.size.width;    //expected width of label
    
    CGRect rect = [self.text boundingRectWithSize:CGSizeMake(finalWidth, finalHeight) options:NSStringDrawingTruncatesLastVisibleLine attributes:attributes context:nil];

    CGSize theStringSize = rect.size;//[self.text sizeWithFont:self.font constrainedToSize:CGSizeMake(finalWidth, finalHeight) lineBreakMode:self.lineBreakMode];
    int newLinesToPad = (finalHeight  - theStringSize.height) / fontSize.height;
    for(int i=0; i<newLinesToPad; i++)
        self.text = [self.text stringByAppendingString:@"\n "];
}

- (void)alignBottom {
    NSDictionary *attributes = @{ NSFontAttributeName: self.font};
    CGSize fontSize = [self.text sizeWithAttributes:attributes];
    double finalHeight = fontSize.height * self.numberOfLines;
    double finalWidth = self.frame.size.width;    //expected width of label
    
    CGRect rect = [self.text boundingRectWithSize:CGSizeMake(finalWidth, finalHeight) options:NSStringDrawingTruncatesLastVisibleLine attributes:attributes context:nil];

    
    CGSize theStringSize = rect.size;//[self.text sizeWithFont:self.font constrainedToSize:CGSizeMake(finalWidth, finalHeight) lineBreakMode:self.lineBreakMode];
    int newLinesToPad = (finalHeight  - theStringSize.height) / fontSize.height;
    for(int i=0; i<newLinesToPad; i++)
        self.text = [NSString stringWithFormat:@" \n%@",self.text];
}
@end

File: Liz/Liz/UILabel+VerticalAlign.swift
import UIKit

extension UILabel {
    
    func alignTop() {
        guard self.text != nil else {
			log.info("Tried to align a label with no text in it.")
            return
        }
        let attributes = [NSFontAttributeName: self.font]
        let fontSize = self.text!.sizeWithAttributes(attributes)
        let finalHeight = fontSize.height * CGFloat(self.numberOfLines)
        let finalWidth = self.frame.size.width
        
        let rect = self.text!.boundingRectWithSize(CGSizeMake(finalWidth, finalHeight), options: .TruncatesLastVisibleLine, attributes: attributes, context: nil)
        let theStringSize = rect.size
        let newLinesToPad = (finalHeight - theStringSize.height)/fontSize.height
        
        Int(newLinesToPad).times {
            self.text = self.text!.stringByAppendingString("\n ")
        }
    }
    
    func alignBottom() {
        guard self.text != nil else {
			log.info("Tried to align a label with no text in it.")
            return
        }
        let attributes = [NSFontAttributeName: self.font]
        let fontSize = self.text!.sizeWithAttributes(attributes)
        let finalHeight = fontSize.height * CGFloat(self.numberOfLines)
        let finalWidth = self.frame.size.width
        
        let rect = self.text!.boundingRectWithSize(CGSizeMake(finalWidth, finalHeight), options: .TruncatesLastVisibleLine, attributes: attributes, context: nil)
        let theStringSize = rect.size
        let newLinesToPad = (finalHeight - theStringSize.height)/fontSize.height
        
        Int(newLinesToPad).times {
            self.text = " \n" + self.text!
        }
    }
}